NFT/src/pages/nft/mint/[id]/[fileId].tsx

```diff
@@ -0,0 +1,131 @@
+import { useRouter } from "next/router";
+
+import { GATEWAY_URL } from "@/lib";
+import { Box, Stack } from "@mui/material";
+import { Button } from "@/components/Button";
+import { useMint } from "@/hooks/useMint";
+import { useTotalSupply } from "@/hooks/useTotalSupply";
+import clsx from "clsx";
+import { NFTImage } from "@/components/NFTImage";
+import { useActiveWallet } from "@/hooks/useActiveWallet";
+import { useGetNFTData } from "@/hooks/useGetNFTData";
+import { Link } from "@/components/Link";
+import { getTruncatedAddress } from "@/utils/address";
+import { Text } from "@/components/Text";
+import { useEffect, useState } from "react";
+
+const MAX_INITIAL_DESCRIPTION = 256;
+
+const NFTDescription = ({ nftDescription }: { nftDescription: string }) => {
+  const [isExpanded, setIsExpanded] = useState(false);
+  const showReadMore = nftDescription.length > MAX_INITIAL_DESCRIPTION;
+
+  return (
+    <>
+      {showReadMore ? (
+        <Stack spacing={2}>
+          {isExpanded ? (
+            <>
+              <Text>{nftDescription}</Text>
+              <Button onClick={() => setIsExpanded(false)}>Show less</Button>
+            </>
+          ) : (
+            <>
+              <Text>{nftDescription.slice(0, MAX_INITIAL_DESCRIPTION)}...</Text>
+              <Button onClick={() => setIsExpanded(true)}>Show more</Button>
+            </>
+          )}
+        </Stack>
+      ) : (
+        <Text>{nftDescription}</Text>
+      )}
+    </>
+  );
+};
+
+export default function Mint() {
+  const router = useRouter();
+  const [minterAddress, setMinterAddress] = useState("");
+
+  const subId = (router.query.nftSubId || "dud") as string;
+  const nftName = router.query.nftName as string;
+  const nftDescription = router.query.nftDescription as string;
+
+  const { totalSupply } = useTotalSupply(subId);
+  const { isConnected } = useActiveWallet();
+
+  const { nftData } = useGetNFTData({
+    keyvalues: {
+      nftSubId: {
+        value: subId,
+        op: "eq",
+      },
+    },
+  });
+
+  useEffect(() => {
+    if (nftData.length && nftData[0].metadata.keyvalues.minter) {
+      setMinterAddress(nftData[0].metadata.keyvalues.minter);
+    }
+  }, [nftData, nftData.length]);
```

Comment:
nftData would ideally be sufficient in case there is any change in the content, might not need `nftData.length`